okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -1460,4 +1563,246 @@ public void alternateService(int streamId, String origin, ByteString protocol, S
       // TODO(madongfly): Deal with alternateService propagation
     }
   }
+
+  /**
+   * SSLSocket wrapper that provides a fake SSLSession for handshake session.
+   */
+  static class SslSocketWrapper extends SSLSocket {
```

Comment:
Make a `NoopSslSocket`, like we did for Netty?

Reply:
Done.